pkg/cmd/server/storage/server.go

```diff
@@ -76,6 +79,102 @@ func (s *StorageRESTServer) Run(stopCh <-chan struct{}) {
 	close(ch)
 }
 
+func (s *StorageRESTServer) sync(key string) (*types.StorageBody, error) {
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
+	sb, ok := s.pushedLocations[key]
+	if ok {
+		return sb, nil
+	}
+	ssb, err := s.st.Fetch(key)
+	return &ssb, err
+}
+
+func (s *StorageRESTServer) del(key string) {
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
+	delete(s.pushedLocations, key)
+}
+
+func (s *StorageRESTServer) handleCatalogCurrentKeySetPost(c *gin.Context) {
```

Comment:
This function does a fair bit - a top-level comment might help for future readers to understand what's exactly being done